Generation/src/components/HepMCFullMerge.h

```diff
@@ -0,0 +1,36 @@
+
+#ifndef GENERATION_HEPMCFULLMERGE_H
+#define GENERATION_HEPMCFULLMERGE_H
+
+#include "FWCore/DataHandle.h"
+#include "Generation/IHepMCMergeTool.h"
+
+#include "GaudiAlg/GaudiTool.h"
+#include "GaudiKernel/RndmGenerators.h" 
+
+#include "HepMC/IO_GenEvent.h"
+#include "HepMC/GenEvent.h"
+
+
+/** 
+ * Implementation partly due to pilemc https://pilemc.hepforge.org/
```

Comment:
Could you also add a line what the difference is w.r.t. the `HepMCSimpleMerge` tool?